clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -148,6 +150,33 @@ public boolean matches(ClientRequest request) {
     }
 
     @Test
+    public void testFetchedRecordsRaisesOnSerializationErrors() {
+        ByteArrayDeserializer deserializer = new ByteArrayDeserializer() {
+            @Override
+            public byte[] deserialize(String topic, byte[] data) {
+                throw new SerializationException();
```

Comment:
That works too. For reference, here is the current handling:
```java
try {
  // parse record
} catch (KafkaException e) {
  throw e;
} catch (RuntimeException e) {
  throw new KafkaException("Error deserializing key/value for partition " + partition + " at offset " + logEntry.offset(), e);
}
```
If we think it's useful to make the partition and offset accessible as fields (maybe it would help if generating an alert for example), then we can do as I suggested above. Otherwise, we probably just need to remove the first catch clause above and maybe change from `KafkaException` to `SerializationException`.